src/structures/paging/frame.rs

```diff
@@ -132,6 +132,23 @@ impl<S: PageSize> Sub<PhysFrame<S>> for PhysFrame<S> {
     }
 }
 
+#[cfg(feature = "step_trait")]
+impl<S: PageSize> core::iter::Step for PhysFrame<S> {
+    fn steps_between(start: &Self, end: &Self) -> Option<usize> {
+        PhysAddr::steps_between(&start.start_address, &end.start_address)
+    }
+
+    fn forward_checked(start: Self, count: usize) -> Option<Self> {
+        PhysAddr::forward_checked(start.start_address, count)
+            .and_then(|start_addr| Self::from_start_address(start_addr).ok())
+    }
+
+    fn backward_checked(start: Self, count: usize) -> Option<Self> {
+        PhysAddr::backward_checked(start.start_address, count)
+            .and_then(|start_addr| Self::from_start_address(start_addr).ok())
+    }
+}
+
```

Comment:
These implementations are incorrect. We need to be taking the page size into account (so that we are consistent with the `PhysFrame<S>: Add<u64>` implementation).

Specifically, `steps_between` should be:
```rust
end.checked_sub(start)?.try_into() 
```
and `forward_checked` should be:
```rust
start.checked_add(count.try_into()?)
```
(similar for `backward_checked`)